chainer/functions/activation/slstm.py

```diff
@@ -176,12 +205,126 @@ def backward(self, inputs, grad_outputs):
                 'lstm_bwd', preamble=_preamble)(
                     c_prev1, a1, i1, f1, o1,
                     c_prev2, a2, i2, f2, o2,
-                    self.c, gc, gh,
+                    c_next, gc, gh,
                     gc_prev1, ga1, gi1, gf1, go1,
                     gc_prev2, ga2, gi2, gf2, go2)
 
         return gc_prev1, gc_prev2, gx1, gx2
 
+    def backward(self, inputs, grads):
+
+        xp = cuda.get_array_module(*inputs)
+
+        c_prev1, c_prev2, x1, x2, c, gc, gh = inputs
+        ggc_prev1, ggc_prev2, ggx1, ggx2 = grads
+
+        if gc is None:
+            gc = xp.zeros_like(c)
+        if gh is None:
+            gh = xp.zeros_like(c)
+        if ggc_prev1 is None:
+            ggc_prev1 = xp.zeros_like(c_prev1)
+        if ggc_prev2 is None:
+            ggc_prev2 = xp.zeros_like(c_prev2)
+
+        gc_prev1 = xp.empty_like(c_prev1)
+        gc_prev2 = xp.empty_like(c_prev2)
+        gx1 = xp.empty_like(x1)
+        gx2 = xp.empty_like(x2)
+        gc_next = xp.empty_like(c)
+        ggc = xp.empty_like(ggc_prev1)
+        ggh = xp.empty_like(gh)
+
+        a1, i1, f1, o1 = _extract_gates(x1)
+        a2, i2, f2, o2 = _extract_gates(x2)
+        gga1, ggi1, ggf1, ggo1 = _extract_gates(ggx1)
+        gga2, ggi2, ggf2, ggo2 = _extract_gates(ggx2)
+        ga1, gi1, gf1, go1 = _extract_gates(gx1)
+        ga2, gi2, gf2, go2 = _extract_gates(gx2)
+
+        o = o1 + o2
+
+        gc_prev1[:], ga1[:], gi1[:], gf1[:], go1[:], \
+            gc_prev2[:], ga2[:], gi2[:], gf2[:], go2[:], \
+            gc_next[:], ggc[:], ggh[:] \
+            = slstm_grad_grad(c_prev1, a1, i1, f1,
+                              c_prev2, a2, i2, f2, o, c, gc, gh,
+                              ggc_prev1, gga1, ggi1, ggf1, ggo1,
+                              ggc_prev2, gga2, ggi2, ggf2, ggo2)
+        return gc_prev1, gc_prev2, gx1, gx2, gc_next, ggc, ggh
+
+
+def _cupy_sigmoid(x):
+    half = x.dtype.type(0.5)
+    return cuda.fusion.tanh(x * half) * half + half
```

Comment:
GPU tests failed on this line: `AttributeError: 'module' object has no attribute 'fusion'`